kernel/src/hil/wifi_cyw43.rs

```diff
@@ -0,0 +1,184 @@
+}
+
+/// Client trait for when the WiFi is initialized and the device
+/// returns the MAC address
+pub trait WifiCtrlClient {
+    /// WiFi device initialized
+    ///
+    /// ## Arguments
+    ///
+    /// - `err`: Status of the initialisation:
+    ///     - `Ok([u8; 6])`if the operation was successful. The buffer
+    ///     provided contains the MAC address of the device
+    ///     - `Err(ErrorCode::FAIL)` if an error occured during initialisation
+    fn init_done(&self, err: Result<[u8; 6], ErrorCode>);
+}
+
+/// Client trait for when the station completes joining a network
+pub trait StationClient {
+    /// Join completed successfully or an error occured
+    ///
+    /// ## Arguments
+    ///
+    /// - `err`: Status of the join operation:
+    ///     - `Ok(())`if the device joined the requested network
+    ///     - `Err(ErrorCode::FAIL)` if an error occured while joining the network
+    fn join_done(&self, err: Result<(), ErrorCode>);
+}
+
+/// Security method
+#[derive(Copy, Clone, Debug, Default)]
+pub enum Security {
+    Wpa,
+    Wpa2,
+    Wpa3,
+    #[default]
+    Wpa2Wpa3,
+}
+
+/// Maximum SSID length
+pub const SSID_SIZE: usize = 32;
+/// Maximum passphrase length
+pub const PS_SIZE: usize = 63;
+
+/// WiFi SSID
+#[derive(Clone, Copy, Debug, Default)]
+pub struct Ssid {
+    pub buf: [u8; SSID_SIZE],
+    pub len: u8,
+}
+
+/// WiFi passphrase
+#[derive(Clone, Copy, Debug)]
+pub struct Passphrase {
+    pub buf: [u8; PS_SIZE],
+    pub len: u8,
+}
+
+impl Default for Passphrase {
+    fn default() -> Self {
+        Self {
+            buf: [0u8; 63],
+            len: Default::default(),
+        }
+    }
+}
+
+/// Join or leave a network with WiFi device configured as a station
+pub trait Station<'a> {
+    /// Set station client
+    fn set_client(&self, client: &'a dyn StationClient);
+
+    /// Join a network, either open or protected
+    ///
+    /// ## Arguments
+    ///
+    /// - `ssid`: WiFi network SSID
+    /// - `security`: Security method to use in order to join the network:
+    ///     - `None` if the network is open
+    ///     - tuple of the security method and the passphrase buffer
+    fn join(&self, ssid: Ssid, security: Option<(Security, Passphrase)>) -> Result<(), ErrorCode>;
+
+    /// Disconnect from the current network
+    fn leave(&self) -> Result<(), ErrorCode>;
+}
+
+/// Scan networks in range
+pub trait Scanner<'a> {
+    /// Set scanner client
+    fn set_client(&self, client: &'a dyn ScannerClient);
+
+    /// Start scanning the available WiFi networks
+    fn start_scan(&self) -> Result<(), ErrorCode>;
+
+    /// Stop scanning
+    fn stop_scan(&self) -> Result<(), ErrorCode>;
+}
+
+/// Client trait for when the scanner found a network
+/// or when the scanning is over
+pub trait ScannerClient {
+    /// Scanner found a network
+    ///
+    /// ## Arguments
+    ///
+    /// - `ssid`: SSID of found network
+    fn scanned_network(&self, ssid: Ssid);
```

Comment:
```suggestion
    fn scanned_network(&self, ssid: Ssid, encryption: Option<Encryption>, signal_strength: u8);
```

```rust
#[non_exhaustive]
enum Encryption {
   Open,
   Wep,
   Wpa1,
   Wpa2,
   Wpa3,
}
```

Just to make it more generic.